source/javascripts/lib/wirsing/filters/search.coffee

```diff
@@ -1,24 +1,37 @@
 #= require ../module
 
-# Returns a configurable search function that can be used
-# to filter an array of objects by searching for a term
-# within the specified property of each object.
-
 wirsing.module 'filter.search', ->
 
+  # Returns a configurable search function that can be used
+  # to filter an array of objects by searching for a term
+  # within the specified property of each object.
+
   (options) ->
 
+    # This is the function that is returned to the outside world.
+    # It takes an array of objects and returns an array of matching objects
     search = (items) ->
       if search.term.length < search.minChars
         items # dont filter
       else
-        items.filter (item) ->
-          item[search.in].toLowerCase().search(search.term.toLowerCase()) > -1
+        items.filter (item) -> search.in.some (key) ->
+          text = removeHyphens item[key]
```

Comment:
So I guess `some` is like `any?` in Ruby and checks if the callback function returns true for any member of the collection, in which case it returns true itself. Correct?